src/System.IO.FileSystem/tests/File/Create.cs

```diff
@@ -165,15 +165,9 @@ public void LongPathSegment()
             }
             else
             {
-                AssertExtensions.ThrowsAny<IOException, PathTooLongException>(
+                AssertExtensions.ThrowsAny<DirectoryNotFoundException, FileNotFoundException, PathTooLongException>(
```

Comment:
Ugh, apparently it is DirectoryNotFoundException, IOException, and PathTooLongException. Sorry.